Frameworks/Foundation/NSUserDefaults.mm

```diff
@@ -232,8 +302,32 @@ - (double)doubleForKey:(NSString*)defaultName {
                                                      ([number isKindOfClass:[NSNumber class]] ? [number doubleValue] : 0.0);
 }
 
+- (void) _scheduleSynchronize {
+    // Up to 2 synchronize operations are allowed in the queue, so that if an existing operation has not been removed from the queue,
+    // we still get a synchronize after the CFPreferencesSetAppValue call. Any additional operations would result in an extra synchronize.
+    if ([_synchronizeQueue operationCount] < 2) {
+        [_synchronizeQueue addOperationWithBlock:^void(void) {
+            [self synchronize];
+        }];
+    }
+}
+
+- (NSOperationQueue*) _synchronizeQueue {
+    return _synchronizeQueue;
+}
+
+- (void) _suspendSynchronize {
+    [_synchronizeQueue setSuspended:YES];
+    [_synchronizeQueue waitUntilAllOperationsAreFinished];
+}
```

Comment:
What happens if we suspend it, then another thread injects a synchronize, and then we try to wait on an operation that'll never complete (since it was suspended)?